### PR TITLE
Unneeded underscore

### PR DESCRIPTION
**Pull request by rrosenblum**
This cop took me much longer to write than I thought it would, but I am always happy with a new cop that fixes offenses inside the RuboCop project.

### CODE REVIEW

**Comment by bbatsov** [comment]
Technically speaking, it might make sense to remove all trailing variables starting with `_`. Maybe this should be some config.

File: config/enabled.yml [hunk 1]
```diff
@@ -714,6 +714,12 @@ Style/UnneededPercentQ:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#percent-q'
   Enabled: true
 
+Style/UnneededUnderscore:
```
Comment (bbatsov) [inline config/enabled.yml]: The name of the cop is pretty confusing. I'm not sure what would be a better name those - maybe something like `RedundantVariable` or something. @jonas054 ideas welcome!

Comment (jonas054) [inline config/enabled.yml]: I think we should prepare for making this cop configurable in the future. Some people might say, "Ending a list of variables with a comma is ugly. I want to enforce the use of underscores.". So the name should not reflect any opinion about whether to use the underscores or not.

Some suggestions:
* `AnonymousUnusedVariable`
* `ThrowawayVariable`
* `UnderscroreVariable`

**Comment by rrosenblum** [comment]
I that it makes sense to add a configuration for removing all trailing underscore variables. What is your thought on the default configuration? Should it remove underscore variables and variables that start with underscore or just underscore variables?

**Comment by rrosenblum** [comment]
Also, what do you think of the cop name being `UnneededVariables`?

**Comment by jonas054** [comment]
@rrosenblum I think that variables that start with underscore, e.g., `_receiver`, is a completely different thing. They are there for documentation. So this cop should not concern itself with them IMO.

In case you want to add a configuration parameter (not sure if it's necessary right now), the default behavior should be the one you've implemented.

**Comment by bbatsov** [comment]
Seems that be build is failing.

**Comment by rrosenblum** [comment]
Sorry about that. The build is passing again.

**Comment by blazeeboy** [comment]
I and @Deradon noticed that no much discussion in here about this cop, we encounter this cop violation in code for a line similar to this :

```
return_value, _ = array_returning_method(arg1, arg2)
```

according to this cop it should look that way

```
return_value, = array_returning_method(arg1, arg2)
```

for me the `return_value,` part looks not so readable like the previous `return_value, _` as the `, _` part is verbose enough to stress on the fact that we have another returning value and we are ignoring it.

are we sure that this code is  aiming to a more readable/better code? 

for me i suggest that multiple `,_ ,_ ,_` should be converted to `, *_` that is verbose enough to say we're ignoring the rest of the assignments.

**Comment by rrosenblum** [comment]
@blazeeboy, I completely understand your concerns about this cop. The main idea is to not include a trailing `_` because `_` by itself provides very little context to the reader. The preferred manor of writing this would be to give a name to the trailing underscore variable. `return_value, _some_other_value = array_returning_method(arg1, arg2)`. If a variable isn't important enough to have a name, then it isn't important enough to be included in assignment.


> i suggest that multiple `,_ ,_ ,_` should be converted to , `*_`

I like the idea of this. We would have to check that there isn't already a splat variable in the variable list before registering an offense or auto-correcting.

**Pull request merged**